src/halmos/exceptions.py

```diff
@@ -20,6 +20,22 @@ class NotConcreteError(HalmosException):
     pass
 
 
+class HalmosTargetReached(Exception):
```

Comment:
please feel free to suggest a better name

Reply:
it sounds a little too semantically charged for me, I would go for something more descriptive like `RunEndingException(Exception)`, and I would make both `HalmosException` and `FailCheatcode` inherit from `RunEndingException`:

```py
class RunEndingException(Exception):
    """
    Base class for any exception that should not just stop the current EVM context, instead aborting the current test run. 
    """

    pass


class HalmosException(RunEndingException):
    """
    Base class for unexpected internal errors happening during a test run. 
    Inherits from RunEndingException because it should not just stop the current EVM context, but stop the whole run. 
    """

    pass


class FailCheatcode(RunEndingException):
    """
    Raised when invoking DSTest's fail() pseudo-cheatcode.
    Inherits from RunEndingException because it should not just stop the current EVM context, but stop the whole run. 
    """

    pass
```

Reply:
some alternatives along the same idea:

- AbortCallException 
- AbortTestException
- RunHaltException
- TestStopException

Reply:
thanks for suggestion! i ended up with `PathEndingException`, because "run" is ambiguous with the entire test run.

Reply:
nice, love it ♥️